radio/src/lua/api_general.cpp

```diff
@@ -1593,6 +1597,50 @@ static int luaSerialWrite(lua_State * L)
   return 0;
 }
 
+/*luadoc
+@function serialRead([num])
+@param num (optional): maximum number of bytes to read.
+                       If non-zero, serialRead will read up to num characters from the buffer.
+                       If 0 or left out, serialRead will read up to and including the first newline character or the end of the buffer.
+                       Note that the returned string may not end in a newline if this character is not present in the buffer.
+
+@retval str string. Empty if no new characters were available.
+
+Reads characters from the serial port. The string is allowed to contain any character, including 0.
+
+@status current Introduced in TODO
+*/
+static int luaSerialRead(lua_State * L)
+{
+  int num = luaL_optunsigned(L, 1, 0);
+
+#if defined(LUA) && !defined(CLI)
+  if (!luaRxFifo) {
+    luaRxFifo = new Fifo<uint8_t, LUA_FIFO_SIZE>();
+    if (!luaRxFifo) {
+      lua_pushlstring(L, "", 0);
+      return 1;
+    }
+  }
+  uint8_t str[LUA_FIFO_SIZE];
+  uint8_t *p = str;
+  while (luaRxFifo->pop(*p)) {
+    p++;  // increment only when pop was successful
+    if (num == 0 && (*(p - 1) == '\n' || *(p - 1) == '\r')) {
+      break;  // found newline
+    }
+    if (p - str >= num || p - str >= LUA_FIFO_SIZE) {
+      break;  // num reached or buffer full
```

Comment:
Still not good, it won't work with num == 0 (you want a full line)

Reply:
Wow I'm not sure how I missed that :S

I've pushed a new fix, hope it is correct this time.